CODING-CONVENTIONS.md

```diff
@@ -0,0 +1,220 @@
+
+  
+# 4 Java 
+
+## 4.1 Style Guide 
+
+Pantheon follows the [Google code style](https://google.github.io/styleguide/javaguide.html) and uses spotless to ensure consistency of formatting. 
+
+To automatically reformat the code before creating a pull request, run: 
+
+```json
+./gradlew spotlessApply
+```
+
+### 4.1.1 Install Google Style Settings 
+
+The Google style settings can be installed in [Intellij](https://github.com/google/google-java-format#intellij) and [Eclipse](https://github.com/google/google-java-format#eclipse). 
+
+## 4.2 Additional Java Style Guidelines
+
+## 4.2.1 Fields 
+
+Class-level fields are generally not separated by blank lines but can use a blank line to separate them into logical groupings.
+
+## 4.2.2 Final Keyword
+
+Method parameters must be final.  Class level and local fields should be final whenever possible.
+
+## 4.2.3 Common Methods
+
+* Getters follow idiomatic format with `get` prefix.  For example, `getBlock()` gets a block property.
+* Setters follow idiomatic format with `set` prefix. For example, `setBlock(Block block)` sets a block property.
+* For `toString methods`,  use the Guava 18+ `MoreObjects.toStringHelper`
+* Equals and `hashCode()` methods use the `Object.equals` and `Object.hash` methods (this is the _Java 7+_ template in IntelliJ. Don’t accept subclasses and don’t use getters)
+
+## 4.2.4 Testing
+
+* Don't use a fixed prefix (for example, `test`). Do explain the expected behaviour not just the situation
+  
+  Good: `returnTrueWhenValueIsXyz()`
+
+  Bad: `valueIsXyz()`
+
+* Use AssertJ for assertions in preference to JUnit’s Assert class.
+
+  To help future-proof the code (avoiding libraries that may be deprecated in the near future), avoid assertions from the `org.assertj.core.api.Java6Assertions` class. Java6 in the name is a concerning signal
+
+## 4.2.5 Miscellaneous 
+
+* When creating loggers it should be the first declaration in the class with: 
+
+  `private static final Logger LOG = getLogger();`
+
+* Ternary operators are acceptable when they make the code clearer but should never be nested
+ 
+* Avoid TODO comments. Log a ticket instead
+
+* Specify Gradle dependency versions in `versions.gradle`
+
+* Don't use two or more blank lines in a row
+
+# 5 Logging
+
+Logging is important for understanding what Pantheon is doing at any given time (for example, progress while synchronizing) and investigating defects.  During development, add logging to aid in these cases. 
+
+## 5.1 Log Messages
+
+Make log messages: 
+* Not so frequent they are overwhelming in the log output 
+* At the appropriate level
+* Detailed enough to understand what actually happened. For example: 
+   
+   `Insufficient validators. Expected 10 but found 4`
+   
+* As succinct as possible while still being clear. 
+
+   * Bad: `Insufficient validators. Expected 10 but got: [address, address, address, address, address, address]`
```

Comment:
No!  This is an excellent.   This should absolutely be there.  It is far easier to track down what is missing when the data is printed in the log message.